pandas/plotting/_core.py

```diff
@@ -1980,50 +1980,160 @@ def plot_series(data, kind='line', ax=None,                    # Series unique
+    ax : object of class matplotlib.axes.Axes, optional
+        The matplotlib axes to be used by boxplot.
+    fontsize : float or str
+        Tick label font size in points or as a string (e.g., `large`).
+    rot : int or float, default 0
+        The rotation angle of labels (in degrees)
+        with respect to the screen coordinate sytem.
+    grid : boolean, default True
+        Setting this to True will show the grid.
     figsize : A tuple (width, height) in inches
-    grid : Setting this to True will show the grid
-    layout : tuple (optional)
-        (rows, columns) for the layout of the plot
-    return_type : {None, 'axes', 'dict', 'both'}, default None
-        The kind of object to return. The default is ``axes``
-        'axes' returns the matplotlib axes the boxplot is drawn on;
-        'dict' returns a dictionary whose values are the matplotlib
-        Lines of the boxplot;
-        'both' returns a namedtuple with the axes and dict.
-
-        When grouping with ``by``, a Series mapping columns to ``return_type``
-        is returned, unless ``return_type`` is None, in which case a NumPy
-        array of axes is returned with the same shape as ``layout``.
-        See the prose documentation for more.
-
-    `**kwds` : Keyword Arguments
+        The size of the figure to create in matplotlib.
+    layout : tuple (rows, columns), optional
+        For example, (3, 5) will display the subplots
+        using 3 columns and 5 rows, starting from the top-left.
+    return_type : {None, 'axes', 'dict', 'both'}, default 'axes'
+        The kind of object to return. The default is ``axes``.
+
+        * 'axes' returns the matplotlib axes the boxplot is drawn on.
+        * 'dict' returns a dictionary whose values are the matplotlib
+          Lines of the boxplot.
+        * 'both' returns a namedtuple with the axes and dict.
+        * when grouping with ``by``, a Series mapping columns to
+          ``return_type`` is returned.
+
+          If ``return_type`` is `None`, a NumPy array
+          of axes with the same shape as ``layout`` is returned.
+    **kwds : Keyword Arguments, optional
         All other plotting keyword arguments to be passed to
-        matplotlib's boxplot function
+        :func:`matplotlib.pyplot.boxplot`.
 
     Returns
     -------
-    lines : dict
-    ax : matplotlib Axes
-    (ax, lines): namedtuple
+    result:
+        Options:
+
+        * ax : object of class
+          matplotlib.axes.Axes (for ``return_type='axes'``)
+        * lines : dict (for ``return_type='dict'``)
+        * (ax, lines): namedtuple (for ``return_type='both'``)
+        * :class:`~pandas.Series` (for ``return_type != None``
+          and data grouped with ``by``)
+        * :class:`~numpy.array` (for ``return_type=None``
+          and data grouped with ``by``)
+
+    See Also
+    --------
+    matplotlib.pyplot.boxplot : Make a box and whisker plot.
+    matplotlib.pyplot.hist : Make a histogram.
```

Comment:
If we consider `hist` a related method, and I agree it is, I'd also add here the pandas version (`Series.plot.hist`).

For the matplotlib ones, I'd use something like "Matplotlib equivalent boxplot", or something like that. I find it slightly confusing that the description is kind of the same as this method, and not explain why.

If we add the pandas `hist`, may be we can leave the matplotlib out, as it should be linked from its page.